source/DomainTheory/Presentation/Type.lagda

```diff
@@ -0,0 +1,77 @@
+(...)
+
+\begin{code}
+{-# OPTIONS --without-K --exact-split --safe --auto-inline #-}
+open import MLTT.Spartan hiding (J)
+
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Subsingletons
+
+module DomainTheory.Presentation.Type
+  (pt : propositional-truncations-exist)
+  (fe : Fun-Ext)
+  {𝓤 𝓣 𝓥 𝓦 : Universe}
+  -- 𝓤 : the universe of the underlying set
+  -- 𝓣 : the universe of the preorder
+  -- 𝓥 : the universe of the indices of directed sets
+  -- 𝓦 : the universe of covering sets
+ where
+
+open import UF.Powerset
+open import Posets.Poset fe
+open PosetAxioms
+
+open import DomainTheory.Basics.Dcpo pt fe 𝓥
+open import DomainTheory.Basics.Miscelanea pt fe 𝓥
+
+module _
+  (G : 𝓤 ̇)  -- Generators
+  (_≲_ : G → G → 𝓣 ̇)
+  where
+
+  Cover-set : 𝓤 ⊔ 𝓥 ⁺ ⊔ 𝓦 ⁺ ̇ -- This one has spurious assumptions
+  Cover-set = G → {I : 𝓥 ̇} → (I → G) → Ω 𝓦
```

Comment:
I am wondering whether this is a workable way to formulate the set of covers in a univalent setting. In particular, I think we would need it to be a family of types rather than a family of propositions. But maybe there is some better formulation anyway.

Reply:
There are two powersets in the original definition. I changed the inner one but not the outer one, because I was not so sure about how the outer one is used.